suite-common/wallet-transactions/src/transactionThunks.ts

```diff
@@ -0,0 +1,222 @@
+                    fee: newFee,
+                    rbf: !!tx.rbf,
+                    blockTime: Math.round(new Date().getTime() / 1000),
+                    // TODO: details: {}, is it worth it?
+                },
+            },
+        };
+        // finalized and recv tx shouldn't have rbfParams
+        if (!tx.rbf || t.tx.type === 'recv') {
+            delete action.payload.tx.rbfParams;
+            return action;
+        }
+
+        if (action.payload.tx.type === 'self') {
+            action.payload.tx.amount = newFee;
+        }
+        // update tx rbfParams
+        if (action.payload.tx.rbfParams) {
+            action.payload.tx.rbfParams = {
+                ...action.payload.tx.rbfParams,
+                txid: newTxid,
+                baseFee: newBaseFee,
+                feeRate: tx.feePerByte,
+            };
+        }
+        return action;
+    });
+    // dispatch replace actions
+    actions.forEach(a => dispatch(a));
+});
+
+export const addFakePendingTxThunk = createAsyncThunk<
+    void,
+    {
+        precomposedTx: Pick<PrecomposedTransactionFinal | TxFinalCardano, 'totalSpent' | 'fee'>;
+        txid: string;
+        account: Account;
+    }
+>(`${modulePrefix}/addFakePendingTransaction`, ({ precomposedTx, txid, account }, { dispatch }) => {
+    // Used in cardano send form and staking tab until Blockfrost supports pending txs on its backend
+    // https://github.com/trezor/trezor-suite/issues/4932
+    const fakeTx = {
+        type: 'sent' as const,
+        txid,
+        blockTime: Math.floor(new Date().getTime() / 1000),
+        blockHash: undefined,
+        // amounts (as most of props below) don't matter much since it is temp fake anyway
+        amount: precomposedTx.totalSpent,
+        fee: precomposedTx.fee,
+        totalSpent: precomposedTx.totalSpent,
+        targets: [],
+        tokens: [],
+        cardanoSpecific: {
+            subtype: null,
+        },
+        details: {
+            vin: [],
+            vout: [],
+            size: 0,
+            totalInput: '0',
+            totalOutput: '0',
+        },
+    };
+    dispatch(transactionActions.addTransaction({ transactions: [fakeTx], account }));
+});
+
+export const prepareFetchTransactionsThunk = ({
+    accountUpdateAction,
+}: {
+    accountUpdateAction: any;
+}) =>
+    createAsyncThunk<
+        void,
+        {
+            account: Account;
+            page: number;
+            perPage: number;
+            noLoading?: boolean;
+            recursive?: boolean;
+        },
+        { state: any }
+    >(
+        `${modulePrefix}/addFakePendingTransaction`,
+        async (
+            { account, page, perPage, noLoading = false, recursive = false },
+            { dispatch, getState },
+        ) => {
+            if (!isTrezorConnectBackendType(account.backendType)) return; // skip unsupported backend type
+            const { transactions } = getState().wallet.transactions;
```

Comment:
never access state directly, getState must be always used in selector

`const transaction = selectTransactions(getState())`

Please fix this everywhere in all thunks

Reply:
https://github.com/trezor/trezor-suite/pull/5981/commits/86973889f127ee510e9e5b3f9fd0d10ad515d1c8